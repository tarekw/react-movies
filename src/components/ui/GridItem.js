import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import Rating from '@material-ui/lab/Rating';

const IMAGE_BASE = 'https://image.tmdb.org/t/p/w300/';

const GridItem = (props) => {
    return (
        <div>
            { props.movie ? (
                <div
                    style={
                        {
                            width: '300px',
                            padding: '10px'
                        }
                    }
                >
                    <Card >
                        <CardMedia
                            style={{height: 450, width: 300}}
                            image={IMAGE_BASE+props.movie.poster_path}
                        />
                        <CardContent>
                            <Rating name="half-rating" value={(props.movie.vote_average / 10) * 5} precision={0.5} />
                            <Typography gutterBottom variant="headline" component="h2">
                                {props.movie.title}
                            </Typography>
                        </CardContent>
                    </Card>
                </div> ) : null
            }
        </div>
    );
};

export default GridItem;
