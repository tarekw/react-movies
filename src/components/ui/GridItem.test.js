import React from 'react';
import { shallow } from 'enzyme';

import GridItem from './GridItem';

const mockMovie = {
	"id": 475557,
	"poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
	"title": "Joker",
	"overview": 'overview',
	"vote_average": 8.7,
};

it('should render correctly with default props', () => {
    const component = shallow(<GridItem />);
    expect(component).toMatchSnapshot();
});

it('should render correctly with mocked props', () => {
    const component = shallow(<GridItem movie={mockMovie} />);
    expect(component).toMatchSnapshot();
});
