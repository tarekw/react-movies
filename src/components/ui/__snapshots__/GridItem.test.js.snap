// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default props 1`] = `<div />`;

exports[`should render correctly with mocked props 1`] = `
<div>
  <div
    style={
      Object {
        "padding": "10px",
        "width": "300px",
      }
    }
  >
    <WithStyles(ForwardRef(Card))>
      <WithStyles(ForwardRef(CardMedia))
        image="https://image.tmdb.org/t/p/w300//udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
        style={
          Object {
            "height": 450,
            "width": 300,
          }
        }
      />
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Rating))
          name="half-rating"
          precision={0.5}
          value={4.35}
        />
        <WithStyles(ForwardRef(Typography))
          component="h2"
          gutterBottom={true}
          variant="headline"
        >
          Joker
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(Card))>
  </div>
</div>
`;
